modules/caddyhttp/opentelemetry/module.go

```diff
@@ -0,0 +1,123 @@
+package opentelemetry
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	"github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(OpenTelemetry{})
+	httpcaddyfile.RegisterHandlerDirective("opentelemetry", parseCaddyfile)
+}
+
+// OpenTelemetry implements an HTTP handler that adds support for the opentelemetry tracing.
+// It is responsible for the injection and propagation of the tracing contexts.
+// OpenTelemetry module can be configured via environment variables https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md. Some values can be overwritten with values from the configuration file.
+type OpenTelemetry struct {
+	// SpanName is a span name. It SHOULD follow the naming guideline https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span
+	SpanName string `json:"span_name"`
```

Comment:
Why not simply `span`?

```suggestion
	SpanName string `json:"span"`
```